xc/device.ml

```diff
@@ -1430,6 +1431,107 @@ let can_surprise_remove ~xs (x: device) = Generic.can_surprise_remove ~xs x
 
 module Dm = struct
 
+  module QMP_Event = struct
+
+    let (pipe_r, pipe_w) = Unix.pipe ()
+
+    module Lookup = struct
+      let ftod, dtoc = Hashtbl.create 16, Hashtbl.create 16
+      let add c domid =
+        Hashtbl.replace ftod (Qmp_protocol.to_fd c) domid;
+        Hashtbl.replace dtoc domid c
+      let remove c domid =
+        Hashtbl.remove ftod (Qmp_protocol.to_fd c);
+        Hashtbl.remove dtoc domid
+      let domid_of fd = try Some (Hashtbl.find ftod fd) with Not_found -> None
+      let channel_of domid = try Some (Hashtbl.find dtoc domid) with Not_found -> None
+    end
+
+    module Epoll = Core.Linux_ext.Epoll
+    module Flags = Core.Linux_ext.Epoll.Flags
+
+    module Monitor = struct
+      let create () = (Core.Std.Or_error.ok_exn Epoll.create) ~num_file_descrs:1001 ~max_ready_events:1001
```

Comment:
I checked the code, `max_ready_events` is used for in epoll_create(max_ready_events), from the manual (man 2 epoll_create), since Linux 2.6.8, the argument is ignored, but must be greater than zero, so I set it as integer 1 here.

Regarding `num_file_descrs`, `Epoll.create` uses `module Bounded_int_table` to store fd, 
from manual:
```
A Bounded_int_table is implemented using two fixed size arrays of size num_keys, which are supplied at table-creation time.
```
Given we support at most 1000 HVMs and need an extra pipe fd, so I set it as integer 1001 here.